### PR TITLE
[ADD] medical_medicament_us and website_sale_medical_medicament_us: Brand/generic IDs and links

### PR DESCRIPTION
**Pull request by obulkin**
- Add the computed fields brand_ids and generic_ids to medical.medicament,
  which return all of the brand-name and generic medicaments with the same GCN
  as the medicament in question, including itself
- Add unit tests for the methods used to compute the new fields
- Extend the website_sale.product template to add links to brand-name
  medicament shop pages that point to equivalent generics and vice versa. These
  are based on the brand_ids and generic_ids fields on medical.medicament
- Add demo data that showcases the change

### CODE REVIEW

**Comment by lasley** [comment]
Travis is failing on lint

File: website_sale_medical_medicament_us/README.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+Configuration
+=============
+
+No configuration is needed or possible.
+
+Usage
+=====
+
+Install and enjoy.
+
```
Comment (lasley) [inline website_sale_medical_medicament_us/README.rst:27]: Missing Runbot link

File: website_sale_medical_medicament_us/__openerp__.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+{
+    'name': 'Medicament Sales - Brand/Generic Links',
+    'summary': 'Links to brand/generic alternatives on medicament shop pages',
+    'version': '9.0.1.0.0',
+    'category': 'Medical',
+    'website': 'https://laslabs.com/',
+    'author': 'LasLabs, Odoo Community Association (OCA)',
+    'license': 'AGPL-3',
+    'application': False,
+    'installable': True,
```
Comment (lasley) [inline website_sale_medical_medicament_us/__openerp__.py:14]: Add `'auto_install': True`. This forces install as long as all dependencies are installed, which is desired for the US modules

**Review by lasley — CHANGES_REQUESTED** [review]

File: medical_medicament_us/models/medical_medicament.py [hunk 3]
```diff
@@ -38,3 +38,29 @@ class MedicalMedicament(models.Model):
...
+    generic_ids = fields.Many2many(
+        help='List of all generic medicaments equivalent to this one',
+        compute='_compute_generic_ids',
+        comodel_name='medical.medicament',
+    )
+
+    @api.multi
+    def _compute_brand_ids(self):
+        for record in self:
+            record.brand_ids = self.env['medical.medicament'].search([
```
Comment (lasley) [inline medical_medicament_us/models/medical_medicament.py]: `self.search`

File: medical_medicament_us/models/medical_medicament.py [hunk 4]
```diff
@@ -38,3 +38,29 @@ class MedicalMedicament(models.Model):
...
+                ('gcn_id.id', '=', record.gcn_id.id),
+                ('gpi', '=', '2'),
+            ])
+
+    @api.multi
+    def _compute_generic_ids(self):
+        for record in self:
+            record.generic_ids = self.env['medical.medicament'].search([
```
Comment (lasley) [inline medical_medicament_us/models/medical_medicament.py]: `self.search`

File: website_sale_medical_medicament_us/demo/medical_medicament.xml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        <field name="drug_form_id" ref="medical_medicament.TAB"/>
+        <field name="gcn_id" ref="acetaminophen_gcn"/>
+        <field name="gpi">2</field>
+        <field name="strength">300</field>
+        <field name="strength_uom_id" ref="medical_medication.product_uom_mg"/>
+        <field name="website_published" eval="True"/>
+    </record>
+
+    <record id="tylenol_brand_2" model="medical.medicament">
+        <field name="name">Tylenol</field>
```
Comment (lasley) [inline website_sale_medical_medicament_us/demo/medical_medicament.xml]: Make this another brand

File: website_sale_medical_medicament_us/demo/medical_medicament.xml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        <field name="drug_form_id" ref="medical_medicament.TAB"/>
+        <field name="gcn_id" ref="acetaminophen_gcn"/>
+        <field name="gpi">2</field>
+        <field name="strength">600</field>
```
Comment (lasley) [inline website_sale_medical_medicament_us/demo/medical_medicament.xml]: GCNs are unique per strength, so this is invalid data. Switch to 300 and change the brand

**Comment by lasley** [comment]
You need to add some security for public and portal user to access medical.medicament.gcn. Log out, then go to a product page using this feature.

You can't access the Runbot logs after initial test, so I paste below:

```
QWebException: (u'Sorry, you are not allowed to access this document. Only users with the following access level are currently allowed to do that:\n- Accounting & Finance/Medical Assistant\n\t- Accounting & Finance/Medical Doctor\n\t- Accounting & Finance/Medical User\n\n(Document model: medical.medicament.gcn)', None)
```

**Comment by obulkin** [comment]
Good catch. I should get in the habit of testing things out when I'm not logged in as admin.

**Comment by obulkin** [comment]
I combined my latest change with the last two PR commits and then remembered that I should instead just add more. My bad. It shouldn't be too annoying here since there's only one change to review.

**Review by lasley — APPROVED** [review]

**Pull request merged**